### PR TITLE
[DS-9408] Enhancing Describe Table & Adding BatchGetItem functionality

### PR DESCRIPTION
**Pull request by Streadz**
### Description
This PR does the following:
* Updates DescribeTable so that Attributes and KeySchemas can be returned
* Adds functionality for performing a batch get

The DescribeTable API is updated because this functionality will be leveraged within our internal DDBQuerier workflow, specifically in the UI where the table properties are hydrated. The hydrated values will be used to make the UI more friendly (i.e. select which attributes you need vs having to explicitly list them out as a user).

The BatchGet functionality is added as a general AWS functionality (and not needed in an RPC call) because it'll be invoked in the backend service component of DDBQuerier for gathering records to be displayed to the user. 

### Testing Performed
Added automated tests that verified client behavior. Additionally I'll be [linking](https://github.com/lyft/clutch-private/#developing) Clutch private to this (while it's in review) and doing additional verification

### CODE REVIEW

**Comment by Streadz** [comment]
👀  #core-datastores-prs @angethuy @dschaller

**Comment by Streadz** [comment]
👀  #core-datastores-prs @angethuy @dschaller

**Comment by Streadz** [comment]
👀  @angethuy

File: api/aws/dynamodb/v1/dynamodb.proto [hunk 1]
```diff
@@ -104,6 +104,12 @@ message Table {
 
   // The aws acccount
   string account = 7;
+
+  // Key schema details for the table
+  repeated KeySchema key_schema = 8;
```
Comment (danielhochman) [inline api/aws/dynamodb/v1/dynamodb.proto:109]:
```suggestion
  repeated KeySchema key_schemas = 8;
```

**Review by angethuy — APPROVED** [review]

File: backend/service/aws/dynamodb.go [hunk 2]
```diff
@@ -307,3 +340,16 @@ func generateIndexUpdates(cl *regionalClient, t *dynamodb.DescribeTableOutput, i
...
+	client, err := c.getAccountRegionClient(account, region)
+	if err != nil {
+		return nil, err
+	}
+	result, err := client.dynamodb.BatchGetItem(ctx, input)
+
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
```
Comment (danielhochman) [inline backend/service/aws/dynamodb.go:354]:
```suggestion
	return client.dynamodb.BatchGetItem(ctx, input)
```

File: api/aws/dynamodb/v1/dynamodb.proto [hunk 3]
```diff
@@ -151,6 +157,28 @@ message Throughput {
   int64 read_capacity_units = 2;
 }
 
+// Represents the Key structure for the table
+message KeySchema {
+  enum Type {
+    // Undetermined value
+    UNSPECIFIED = 0;
```
Comment (danielhochman) [inline api/aws/dynamodb/v1/dynamodb.proto:164]: you should add UNKNOWN for the case when the type cannot be determined coming from the AWS SDK.

**Review by danielhochman — APPROVED** [review]
lgtm, small comments

**Review by miguelmolina95 — APPROVED** [review]

File: api/aws/dynamodb/v1/dynamodb.proto (hunk 3 above)
Comment (Streadz) [reply api/aws/dynamodb/v1/dynamodb.proto:164]: Ok, will do!

**Comment by Streadz** [comment]
💨

**Pull request merged**